lib/puppet/network/http/api/master/v3/environment.rb

```diff
@@ -20,14 +20,13 @@ def call(request, response)
     applications.each do |app|
       app_components = {}
       # Turn the 'nodes' hash into a map component ref => node name
-      node_mapping = app['nodes'].map do |k,v|
-        k.type == "Node" ? [k, v] : [v, k]
-      end.reduce({}) do |map, (node, comp)|
-        unless map[comp.ref].nil?
-          raise Puppet::ParseError, "Application #{app} maps component #{comp} to mutiple nodes"
+      node_mapping = {}
+      app['nodes'].each do |node, comps|
+        comps = [comps] unless comp.is_a?(Array)
```

Comment:
Not sure why. We already check that a node is associated to one or several resources. We don't allow nested arrays in that check. Regarding unique, is it meaningful to hide duplicates in a flat array? As it stands now a duplicate will trigger the 'maps component ... to multiple nodes' just below this comment (since the same component is mapped to the same node twice).